projects/planet-finance/index.js

```diff
@@ -271,48 +237,101 @@ async function tvl(timestamp, ethBlock, chainBlocks) {
     }
   }
 
-  //Green Planet
-
+  //green markets tvl -- starts
   let a = Object.keys(greenMarkets)
   let marketLength = Object.keys(greenMarkets).length;
 
   for (var i = 0; i < marketLength; i++) {
-    let totalSupply = (
-      await sdk.api.abi.call({
+    let totalSupply = 0;
+    let exchangeRateStored = 0;
+    if(Object.keys(greenMarkets)[i] == "0x6bD50dFb39699D2135D987734F4984cd59eD6b53" || Object.keys(greenMarkets)[i] == "0x6E7a174836b2Df12599ecB2Dc64C1F9e1576aC45")
+    {
+      let targetObjectArr = {
+        "0x6bD50dFb39699D2135D987734F4984cd59eD6b53" :  
+          "0x0c6dd143F4b86567d6c21E8ccfD0300f00896442", // iGamma to Gamma
+        "0x6E7a174836b2Df12599ecB2Dc64C1F9e1576aC45":
+          "0xb7eD4A5AF620B52022fb26035C565277035d4FD7"  //  iAqua to Aqua
+      }
+      //console.log(targetObjectArr);
+      //infinity gamma staking
+      //getting gGamma locked from infinity vaults
+      totalSupply = ( await sdk.api.abi.call({
         target: Object.keys(greenMarkets)[i],
-        abi: abi["totalSupply"],
+        abi: abi["balanceOfGtoken"],
         chain: "bsc",
         block: chainBlocks.bsc,
-      })
-    ).output;
+      })).output;
+
+      //gettting exchange rate between Gamma and gGamma
+      exchangeRateStored = (
+        await sdk.api.abi.call({
+          target: targetObjectArr[Object.keys(greenMarkets)[i]],
+          abi: abi["exchangeRateStored"],
+          chain: "bsc",
+          block: chainBlocks.bsc,
+        })
+      ).output;
+    }
+    else
+    {
+      totalSupply = (
+        await sdk.api.abi.call({
+          target: Object.keys(greenMarkets)[i],
+          abi: abi["totalSupply"],
+          chain: "bsc",
+          block: chainBlocks.bsc,
+        })
+      ).output;
+
+      exchangeRateStored = (
+        await sdk.api.abi.call({
+          target: Object.keys(greenMarkets)[i],
+          abi: abi["exchangeRateStored"],
+          chain: "bsc",
+          block: chainBlocks.bsc,
+        })
+      ).output;
+    }
 
-    const exchangeRateStored = (
-      await sdk.api.abi.call({
-        target: Object.keys(greenMarkets)[i],
-        abi: abi["exchangeRateStored"],
-        chain: "bsc",
-        block: chainBlocks.bsc,
-      })
-    ).output;
-      
     let exchangeRateDecimal = 1e18;
     if(Object.keys(greenMarkets)[i] == "0x8b04e56a8cd5f4d465b784ccf564899f30aaf88c")
     {
       exchangeRateDecimal = 1e15
     }
 
+    let addr = greenMarkets[Object.keys(greenMarkets)[i]];
+
+    if(Object.keys(greenMarkets)[i] == "0x6bD50dFb39699D2135D987734F4984cd59eD6b53")
+    {
+      //for adding marketTvl from Gamma infinity (iGamma) to gGamma staking
+      addr = "0xb3Cb6d2f8f2FDe203a022201C81a96c167607F15";
```

Comment:
what are the g tokens? Surely i tokens are worth less? is this a curve style infinite locked token? Can you explain the relationship between GAMMA, gGAMMA and iGAMMA?

Reply:
I've had a read of the docs and it seems like gGAMMA and iGAMMA TVL is double counting the first order deposits?

Reply:
@waynebruce0x 

When you supply GAMMA, you get gGAMMA in return. gGAMMA is a gToken and gTokens generally are interest bearing token. (gGAMMA is not yet an interest bearing token. It will soon be)

User can supply gGAMMA to infinity vault and get iGAMMA in return. User receives rewards again for this. 

It is not double counting as user is supplying different tokens to two different vaults and getting two different rewards.

Reply:
Update regarding Solana and aUST - https://github.com/DefiLlama/DefiLlama-Adapters/pull/2418/commits/9c4c287c497f45ebe2fd479b4cf254e242463637

As Solana and aUST are not listed on coingecko so using other listed tokens (Solana and UST) on coingecko to add to TVL.

For Solana
Adding TVL from this
https://bscscan.com/address/0xfea6ab80cd850c3e63374bc737479aeec0e8b9a1

to this  for Solana
https://bscscan.com/address/0xfa54ff1a158b5189ebba6ae130ced6bbd3aea76e#code

For aUST
Adding TVL from 
https://bscscan.com/address/0x8b04e56a8cd5f4d465b784ccf564899f30aaf88c

to this for aUST. 
https://bscscan.com/address/0x3d4350cd54aef9f9b2c29435e0fa809957b3f30a

Here I am converting marketTvl from aUST to UST.
Also have submitted request at coingecko for listing wormhole aUST.
Will update my code later when it gets listed.

Reply:
Since gGAMMA is a receipt of deposited GAMMA, if we then count the receipt again it's effectively counting the original GAMMA twice. And then it seems like iGAMMA is another order/wrapped above that